src/watchdog_threadpool.c

```diff
@@ -67,27 +67,16 @@ IMPLEMENT_MOCKABLE_FUNCTION(, int, watchdog_threadpool_init)
             }
             else
             {
-                /*Codes_SRS_WATCHDOG_THREADPOOL_42_004: [ watchdog_threadpool_init shall open the threadpool by calling threadpool_open. ]*/
-                if (threadpool_open(threadpool) != 0)
-                {
-                    /*Codes_SRS_WATCHDOG_THREADPOOL_42_007: [ If there are any other errors then watchdog_threadpool_init shall fail and return a non-zero value. ]*/
-                    LogError("threadpool_open failed");
-                    result = MU_FAILURE;
-                }
-                else
-                {
-                    /*Codes_SRS_WATCHDOG_THREADPOOL_42_006: [ watchdog_threadpool_init shall store the threadpool. ]*/
-                    THANDLE_INITIALIZE_MOVE(THREADPOOL)(&g_watchdog.threadpool, &threadpool);
-
-                    g_watchdog.execution_engine = execution_engine;
-
-                    (void)interlocked_exchange(&g_watchdog_init_state, WATCHDOG_INIT_OK);
-
-                    /*Codes_SRS_WATCHDOG_THREADPOOL_42_008: [ watchdog_threadpool_init shall return 0. ]*/
-                    result = 0;
-                    goto all_ok;
-                }
-                THANDLE_ASSIGN(THREADPOOL)(&threadpool, NULL);
+                /*Codes_SRS_WATCHDOG_THREADPOOL_42_006: [ watchdog_threadpool_init shall store the threadpool. ]*/
+                THANDLE_INITIALIZE_MOVE(THREADPOOL)(&g_watchdog.threadpool, &threadpool);
+
+                g_watchdog.execution_engine = execution_engine;

```

Comment:
>g_watchdog.execution_engine = execution_engine; [](http://example.com/codeflow?start=16&length=47)

a long time ago threadpool did not do an inc_ref on the execution_engine, so this module had to do it itself so the execution engine didn't get freed while the threadpool was using it.

But now threadpool does correctly take a reference (inc_ref).

Since this module has no need for the execution_engine after handing it off to threadpool I think we could remove some more dead code by no longer storing this (instead moving the `execution_engine_dec_ref` up here and removing `execution_engine` from the `WATCHDOG_THREADPOOL` struct entirely.

/consider as another small cleanup while touching this.

Reply:
now execution engine is dec_Ref'd only here.